plugins/CoreHome/Controller.php

```diff
@@ -185,10 +187,44 @@ protected function setDateTodayIfWebsiteCreatedToday()
     public function index()
     {
         $this->setDateTodayIfWebsiteCreatedToday();
+
+        if ($this->noRecordedData()) {
+            return $this->showSiteNoData();
+        }
+
         $view = $this->getDefaultIndexView();
         return $view->render();
     }
 
+    private function noRecordedData()
+    {
+        $sql = "SELECT COUNT(*)
+                FROM " . Common::prefixTable('log_visit') . "
+                WHERE idsite=?";
+
+        $count = Db::fetchOne($sql, array($this->idSite));
+
+        return $count == 0;
+    }
```

Comment:
Actually, there is an index on `idsite, visit_last_action_time` and `idsite, idvisitor`. So it would be even better to use for example `SELECT idvisitor FROM piwik_log_visit WHERE idsite=? limit 1`since it then can read from the index which would be even faster